llvm/test/Transforms/LoopVectorize/X86/pr147398.ll

```diff
@@ -0,0 +1,126 @@
+; CHECK-NEXT:    [[TMP25:%.*]] = extractelement <4 x i1> [[TMP18]], i32 3
+; CHECK-NEXT:    br i1 [[TMP25]], label %[[PRED_STORE_IF5:.*]], label %[[PRED_STORE_CONTINUE6]]
+; CHECK:       [[PRED_STORE_IF5]]:
+; CHECK-NEXT:    [[TMP26:%.*]] = extractelement <4 x double> [[PREDPHI]], i32 3
+; CHECK-NEXT:    store double [[TMP26]], ptr null, align 8
+; CHECK-NEXT:    br label %[[PRED_STORE_CONTINUE6]]
+; CHECK:       [[PRED_STORE_CONTINUE6]]:
+; CHECK-NEXT:    [[INDEX_NEXT]] = add nuw i64 [[INDEX]], 4
+; CHECK-NEXT:    [[VEC_IND_NEXT]] = add <4 x i8> [[VEC_IND]], splat (i8 4)
+; CHECK-NEXT:    br i1 true, label %[[MIDDLE_BLOCK:.*]], label %[[VECTOR_BODY]], !llvm.loop [[LOOP0:![0-9]+]]
+; CHECK:       [[MIDDLE_BLOCK]]:
+; CHECK-NEXT:    br label %[[LOOP_EXIT:.*]]
+; CHECK:       [[SCALAR_PH]]:
+; CHECK-NEXT:    [[BC_RESUME_VAL:%.*]] = phi i64 [ 0, %[[START]] ]
+; CHECK-NEXT:    br label %[[LOOP_BODY:.*]]
+; CHECK:       [[LOOP_BODY]]:
+; CHECK-NEXT:    [[INVAR_021:%.*]] = phi i64 [ [[BC_RESUME_VAL]], %[[SCALAR_PH]] ], [ [[INVAR_INC11:%.*]], %[[AFTER:.*]] ]
+; CHECK-NEXT:    br i1 false, label %[[AFTER]], label %[[CHECK_TRUE:.*]]
+; CHECK:       [[LOOP_EXIT]]:
+; CHECK-NEXT:    ret i64 0
+; CHECK:       [[AFTER]]:
+; CHECK-NEXT:    [[RET_VALUE_ADDR_0:%.*]] = phi double [ [[TMP32:%.*]], %[[CHECK_TRUE]] ], [ 0.000000e+00, %[[LOOP_BODY]] ]
+; CHECK-NEXT:    store double [[RET_VALUE_ADDR_0]], ptr null, align 8
+; CHECK-NEXT:    [[INVAR_INC11]] = add i64 [[INVAR_021]], 1
+; CHECK-NEXT:    [[EXITCOND:%.*]] = icmp eq i64 [[INVAR_021]], 1
+; CHECK-NEXT:    br i1 [[EXITCOND]], label %[[LOOP_EXIT]], label %[[LOOP_BODY]], !llvm.loop [[LOOP3:![0-9]+]]
+; CHECK:       [[CHECK_TRUE]]:
+; CHECK-NEXT:    [[TMP27:%.*]] = sub nsw i64 [[INVAR_021]], 0
+; CHECK-NEXT:    [[TMP28:%.*]] = trunc i64 [[TMP27]] to i32
+; CHECK-NEXT:    [[TMP29:%.*]] = icmp sgt i32 0, [[TMP28]]
+; CHECK-NEXT:    [[TMP30:%.*]] = getelementptr [16 x [16 x double]], ptr [[ARG2]], i64 0, i64 [[DIM]], i64 [[TMP27]]
+; CHECK-NEXT:    [[TMP31:%.*]] = load double, ptr [[TMP30]], align 8
+; CHECK-NEXT:    [[TMP32]] = select i1 [[TMP29]], double 1.000000e+00, double 0.000000e+00
+; CHECK-NEXT:    br label %[[AFTER]]
+;
+start:
+  br label %loop_body
+
+loop_body:                                        ; preds = %after, %start
+  %invar.021 = phi i64 [ 0, %start ], [ %invar.inc11, %after ]
+  br i1 false, label %after, label %check_true
+
+loop_exit:                                        ; preds = %after
+  ret i64 0
+
+after:                                            ; preds = %check_true, %loop_body
+  %ret_value_addr.0 = phi double [ %5, %check_true ], [ 0.000000e+00, %loop_body ]
+  store double %ret_value_addr.0, ptr null, align 8
+  %invar.inc11 = add i64 %invar.021, 1
+  %exitcond = icmp eq i64 %invar.021, 1
+  br i1 %exitcond, label %loop_exit, label %loop_body
+
+check_true:                                       ; preds = %loop_body
+  %0 = sub nsw i64 %invar.021, 0
```

Comment:
Better subtract something other than 0, to avoid this being optimized away in the future.